core/trino-main/src/main/java/io/trino/execution/scheduler/policy/PhasedExecutionSchedule.java

```diff
@@ -142,8 +149,16 @@ public boolean isFinished()
     @VisibleForTesting
     void schedule()
     {
-        removeCompletedStages();
-        unblockStagesWithFullOutputBuffer();
+        ImmutableSet.Builder<PlanFragmentId> fragmentsToExecute = ImmutableSet.builder();
+        fragmentsToExecute.addAll(removeCompletedStages());
+        fragmentsToExecute.addAll(unblockStagesWithFullOutputBuffer());
+        selectForExecution(fragmentsToExecute.build());
+    }
+
+    @VisibleForTesting
+    List<PlanFragmentId> getSortedFragments()
+    {
+        return sortedFragments;
```

Comment:
What is the point of asserting on this one. Oredering is arbitrary. And you do assertion on execution order anyway. I would make it private.

Reply:
I want to make sure that global ordering is as expected. Oredering is not arbitrary as it favors build before probe and upstream stage before downstream.

Reply:
It is worth a comment in code then maybe. As code which collects ordering sequence is just an in-order tree scan. It is not obvious that abovementioned properties are important.